lib/hammer_cli/options/option_collector.rb

```diff
@@ -0,0 +1,22 @@
+module HammerCLI
+  module Options
+    class OptionCollector
+      attr_accessor :option_sources
+
+      def initialize(recognised_options, option_sources)
+        @recognised_options = recognised_options
+        @option_sources = option_sources
+      end
+
+      def all_options
+        @all_options ||= @option_sources.inject({}) do |all_options, source|
+          source.get_options(@recognised_options, all_options)
```

Comment:
I want option sources to have access to the full option hash, because there can be situations when you need other options' values. For example whole name -> id resolution could be refactored into an option source.